samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -256,6 +254,10 @@ public ContainerStorageManager(
         containerChangelogSystems, systemFactories, config, this.samzaContainerMetrics.registry());
     this.storeConsumers = createStoreIndexedMap(this.changelogSystemStreams, storeSystemConsumers);
 
+    // TODO HIGH dchen tune based on observed concurrency
+    this.restoreExecutor = Executors.newFixedThreadPool(containerModel.getTasks().size(),
```

Comment:
@dxichen I think this still needs to be 2x until you make RestoreManager init() and restore() nonblocking.

Where is restore manager init (not store init) called btw? Is this on this executor as well? Since we're deleting old stores during restore init, ideally that should be parallel and nonblocking as well.